sale_manual_delivery/wizard/manual_proc.py

```diff
@@ -0,0 +1,166 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
+from odoo.tools import float_compare
+from odoo.exceptions import UserError
+from odoo.tools.translate import _
+
+
+PARTNER_DOMAIN = """[
+    "|", ("id", "=", commercial_partner_id),
+    ("parent_id", "=", commercial_partner_id),
+]"""
+
+
+class ManualDelivery(models.TransientModel):
+    """Creates procurements manually"""
+    _name = "manual.delivery"
+    _order = 'create_date desc'
+
+    @api.model
+    def default_get(self, fields):
+        res = super(ManualDelivery, self).default_get(
+            fields)
+        if 'line_ids' not in fields:
+            return {}
+        res = super(ManualDelivery, self).default_get(fields)
+        active_model = self.env.context['active_model']
+        if active_model == 'sale.order.line':
+            sale_ids = self.env.context['active_ids'] or []
+            lines = self.env['sale.order.line'].browse(sale_ids).filtered(
+                lambda s: s.pending_qty_to_deliver)
+        elif active_model == 'sale.order':
+            sale_ids = self.env.context['active_ids'] or []
+            lines = self.env['sale.order'].browse(sale_ids).mapped(
+                'order_line').filtered(
+                lambda s: s.pending_qty_to_deliver)
+        res['sale_line_ids'] = [(6, 0, lines.ids)]
+        if len(lines.mapped('order_id.partner_id')) > 1:
+            raise UserError(_('Please select one partner at a time'))
+        res['line_ids'] = self.fill_lines(lines)
+        partner = lines.mapped('order_id.partner_id')
+        res['partner_id'] = partner.id
+        res['commercial_partner_id'] = partner.commercial_partner_id.id
+        return res
+
+    @api.multi
+    def fill_lines(self, sale_lines):
+        lines = []
+
+        for line in sale_lines:
+            if (not line.existing_qty == line.product_uom_qty and
+                    line.product_id.type != 'service'):
+                vals = {
+                    'product_id': line.product_id.id,
+                    'line_description': line.product_id.name,
+                    'order_line_id': line.id,
+                    'ordered_qty': line.product_uom_qty,
+                    'existing_qty': line.existing_qty,
+                    'to_ship_qty': line.product_uom_qty - line.existing_qty
+                }
+                lines.append((0, 0, vals))
+        return lines
+
+    @api.onchange('line_ids')
+    def onchange_line_ids(self):
+        lines = []
+        for line in self.sale_line_ids:
+            if (not line.existing_qty == line.product_uom_qty and
+                    line.product_id.type != 'service'):
+                vals = {
+                    'order_line_id': line.id,
+                    'ordered_qty': line.product_uom_qty,
+                    'existing_qty': line.existing_qty,
+                    'to_ship_qty': line.product_uom_qty - line.existing_qty
```

Comment:
no need to take in count qty_delivered?

Reply:
I think existing_qty is already considered the delivered_qty. I will check.

Reply:
It is